hypex/executor/calculators.py

```
@@ -221,112 +210,58 @@ def _inner_function(
-        power_iteration_size: int = 3000,
-        alpha: float = 0.05,
-        iteration_size: int = 5000,
-        equal_variance: bool = True,
-        random_state: int | None = 42,
-        **kwargs,
-    ) -> dict:
-        group_field = Adapter.to_list(group_field)
-
-        if grouping_data is None:
-            grouping_data = data.groupby(group_field)
-
-        if len(grouping_data) <= 1:
-            raise NotSuitableFieldError(group_field, "Grouping")
-
-        if variances is None:
-            group_vars = cls._variance_by_group(grouping_data, target_field=target_field)
-            if equal_variance:
-                variances_used: list[float] | float = float(np.mean(group_vars))
-            else:
-                variances_used = group_vars
-        else:
-            variances_used = variances
-
-        num_samples = len(grouping_data)
-
-        res = cls._inner_function(
-            num_samples=num_samples,
-            mde=mde,
-            variances=variances_used,
-            power=power,
-            quantile_1=quantile_1,
-            quantile_2=quantile_2,
-            initial_estimate=initial_estimate,
-            power_iteration_size=power_iteration_size,
-            alpha=alpha,
-            iteration_size=iteration_size,
-            equal_variance=equal_variance,
-            random_state=random_state,
-        )
-
-        payload = {
-            "min sample size": res,
-        }
-       
-        return {target_field: payload}
-
-    def execute(self, data: Dataset) -> dict:
-        data = ExperimentData(data)
+    def calc(self, data: Dataset) -> dict:
         group_field, target_fields = self._get_fields(data=data)
 
-        self.key = str(
-            target_fields[0] if len(target_fields) == 1 else (target_fields or "")
-        )
+        self.key = str(target_fields[0] if len(target_fields) == 1 else (target_fields or ""))
+
+        if not target_fields and data.tmp_roles:
+            return {}
```

it should raise error instead of returning empty dict